net/rootd/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@
 ############################################################################
 
 ROOT_EXECUTABLE(rootd *.cxx ${CMAKE_SOURCE_DIR}/core/clib/src/strlcpy.c
-  LIBRARIES rpdutil rsa ${GLOBUS_LIBRARIES} ${OPENSSL_LIBRARIES})
+  LIBRARIES rpdutil rsa ${GLOBUS_LIBRARIES} ${OPENSSL_LIBRARIES}
+  BUILTINS OPENSSL)
```

Comment:
I don't understand, how does this add libcrypto?

Reply:
This was the only place where it said LIBRARIES ${OPENSSL_LIBRARIES} without BUILTINS OPENSSL. You should always have both. ${OPENSSL_LIBRARIES} provides libssl and libcrypto when the system libraries are used, BUILTINS OPENSSL provides libssl and libcrypto when the builtin openssl is used. Having {OPENSSL_LIBRARIES} without BUILTINS OPENSSL is inconsistent.

Reply:
I updated the builtin for OpenSSL, so that having `BUILTINS OPENSSL` would not be necessary anymore. Is there a way I can reproduce the error you see? See, for example, commits  65bbb7652f6294e74d328f3bc8adf6690d1e9967 and 3787d16a54f3b626812d0b5c412d002c1784031c, and take a look at the log for `builtins/openssl` in  ROOT for more details. If some configuration doesn't work for you, I'd like to fix the `CMakeLists.txt` for builtin OpenSSL. Eventually, I want the `BUILTINS` option to disappear from the build system. The dependencies should be there by default regardless if a package is builtin or not.

Reply:
The change here was just for consistency.
The change needed to fix the link error was the change in the rule for compiling libSrvAuth.so in net/rpdutils/CMakeLists.txt. After the change that made the dependency on openssl in libNet private, this dependency is no longer inherited by SrvAuth by declaring DEPENDENCIES Net. And since SrvAuth uses openssl directly the linking started to fail.

Reply:
The undefined references come from globus.cxx.o, so in order to reproduce you need to configure using -Dglobus:BOOL=ON.

Reply:
Thanks. I will try with this configuration and fix the problem. The `BUILTINS` option will gradually go away, so I don't want to add it back if it's not needed.